internal/preferences.go

```diff
@@ -67,12 +72,15 @@ func (p *InMemoryPreferences) remove(key string) {
 }
 
 func (p *InMemoryPreferences) fireChange() {
-	p.lock.Lock()
-	defer p.lock.Unlock()
+	p.lock.RLock()
+	defer p.lock.RUnlock()
 
 	for _, l := range p.changeListeners {
+		p.waitGroup.Add(1)
```

Comment:
It would seem cleaner (assuming it's not to big an overhead) to put the Done inside an anonymous goroutine that then calls `l()` here so that we have code symmetry and safety against the potential issues I mentioned above.

Reply:
I understood how this will make the code in fact much clearer, but i didn't get what was the issue exactly, can you please elaborate?

Reply:
When operations happen in pairs (`Add`, `Done` or `Open` and `Close`) it is desirable to keep them together so that it is clear that resources are freed. By separating these it is easy for bugs to be introduced or for debugging to take longer.

Reply:
Got it, thanks
I will make the changes asap